rgw/v2/tests/s3_swift/test_quota_management.py

```diff
@@ -0,0 +1,209 @@
+import argparse
+import traceback
+
+import v2.lib.resource_op as s3lib
+import v2.utils.utils as utils
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.resource_op import Config
+from v2.lib.s3.auth import Auth
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.tests.s3_swift import reusable
+from v2.tests.s3_swift.reusables import quota_management as quota_mgmt
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+
+TEST_DATA_PATH = None
+import logging
+
+log = logging.getLogger()
+
+
+def upload_object_initiate(
+    config, each_user, bucket, bucket_name, obj_name_suffix, obj_size
+):
+    config.obj_size = obj_size
+    s3_object_name = utils.gen_s3_object_name(bucket_name, obj_name_suffix)
+    log.info("s3 object name: %s" % s3_object_name)
+    s3_object_path = os.path.join(TEST_DATA_PATH, s3_object_name)
+    log.info("s3 object path: %s" % s3_object_path)
+    log.info("upload type: normal")
+    try:
+        reusable.upload_object(
+            s3_object_name,
+            bucket,
+            TEST_DATA_PATH,
+            config,
+            each_user,
+        )
+        return True
+    except TestExecError as e:
+        log.info(e)
+        return False
+
+
+def test_max_objects(quota_scope, config, each_user, bucket, max_objects):
+    log.info(f"testing {quota_scope} quota max objects")
+    quota_mgmt.set_quota(
+        quota_scope=quota_scope, user_info=each_user, max_objects=max_objects
+    )
+    quota_mgmt.toggle_quota("enable", quota_scope, each_user)
+
+    log.info(f"uploading {max_objects} objects equal to quota limit")
+    for i in range(1, max_objects + 1):
+        uploaded = upload_object_initiate(config, each_user, bucket, bucket.name, i, 0)
+        if not uploaded:
+            AssertionError(
+                f"{quota_scope} quota with max objects failed as upload object refused before reaching limit"
+            )
+    log.info(
+        f"uploading one more objects to test {quota_scope} quota max objects limit"
+    )
+    uploaded = upload_object_initiate(config, each_user, bucket, bucket.name, i + 1, 0)
+    if uploaded:
+        raise AssertionError(f"{quota_scope} quota with max objects failed")
+    log.info(f"object upload failed as {quota_scope} quota max objects limit exceeded")
+
+    quota_mgmt.toggle_quota("disable", quota_scope, each_user)
+    log.info(f"uploading one more object after disabling {quota_scope} quota")
+    uploaded = upload_object_initiate(config, each_user, bucket, bucket.name, i + 1, 0)
+    if not uploaded:
+        raise AssertionError(
+            f"object upload refused even after disabling {quota_scope} quota"
+        )
+    log.info("object upload successful after disabling quota")
+    reusable.delete_objects(bucket)
+    log.info(f"{quota_scope} quota with max objects passed")
+
+
+def test_max_size(quota_scope, config, each_user, bucket, max_size):
```

Comment:
@hmaheswa , could we move this also to resuables/quota_management.py

Reply:
done madhavi